src/overlay/Peer.cpp

```diff
@@ -1029,12 +1063,61 @@ Peer::addMsgAndMaybeTrimQueue(std::shared_ptr<StellarMessage const> msg)
             }
         }
     }
+    else if (type == FLOOD_ADVERT)
+    {
+        uint32_t limit = mApp.getLedgerManager().getLastMaxTxSetSizeOps() /
```

Comment:
Also this doesn't seem right: what happens if we have many adverts that contain small numbers of hashes? Then the limit is too aggressive. Instead, you can track the number of hashes in the queue, and trim the queue based on that

Reply:
Fixed.